src/commands/lambda/flare.ts

```diff
@@ -365,9 +431,20 @@ export const getLogStreamNames = async (cwlClient: CloudWatchLogsClient, logGrou
   const output: string[] = []
   for (const logStream of logStreams) {
     const logStreamName = logStream.logStreamName
-    if (logStreamName) {
-      output.push(logStreamName)
+    if (!logStreamName) {
+      continue
+    }
+    if (rangeSpecified) {
+      const lastEventTime = logStream.lastEventTimestamp
+      const firstEventTime = logStream.firstEventTimestamp
+      if (!lastEventTime || !firstEventTime) {
+        continue
+      }
+      if (lastEventTime < startMillis || firstEventTime > endMillis) {
+        continue
```

Comment:
`DescribeLogStreamsCommand` doesn't accept a start or end time, so i had to filter log streams manually